### ISSUE TITLE
Search: Add taxonomy terms to search results 

### ISSUE THREAD

**Issue by htdat** [Automattic/jetpack#14790]
#### Is your feature request related to a problem? Please describe.

Currently, search results include posts only. If possible, we add taxonomy terms to results too.

#### Describe the solution you'd like

Lets say I have custom taxonomy „cars“ and it contains terms like „bmw“, „audi“,“mercedes“,“mercedes-benz“ …

When user on my site is searching for query „mercedes“, it will show him (besides the found posts) links to the terms „mercedes“ and “mercedes-benz“.

So basically it would index the terms and use them for search too.

**Comment by gibrown** [comment]
This is something we have considered and talked about, so it is great to hear a user actually requesting it.
